movie_library/views/actor_controller.py

```diff
@@ -1,27 +1,74 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import status
 from rest_framework.viewsets import ViewSet
 from rest_framework.response import Response
+from ..utils.custome_premissions import CustomPermissions
+from ..models.movie import Movie
 from ..models.actor import Actor
 from ..utils.serializers.actor_serializer import ActorSerializer
+from ..utils.serializers.actor_movies_serializer import ActorMoviesSerializer
+from ..models.movie_actor import MovieActor
 
 
 class ActorViewSet(ViewSet):
+    permission_classes = [CustomPermissions]
 
     def list(self, request):
         actors = Actor.objects.all()
         result = ActorSerializer(actors, many=True)
         return Response(data=result.data, status=200)
 
     def create(self, request):
-        pass
+        data = request.data
+        movies = data.get('movies')
+        actor = Actor(first_name=data.get('first_name'), last_name=data.get('last_name'), gender=data.get('gender'))
+        actor.save()
+        bulk_list = list()
+        query = Movie.objects.all()
+        for movie in movies:
+            the_movie = get_object_or_404(query, pk=movie)
+            #the_movie = Movie.objects.filter(pk=movie)
+            bulk_list.append(
+                MovieActor(actors=actor, movies=the_movie))
+
+        bulk_msj = MovieActor.objects.bulk_create(bulk_list)
+        result = ActorMoviesSerializer(actor, many=False)
+        return Response(data=result.data, status=201)
 
     def retrieve(self, request, pk=None):
-        pass
+        actors = Actor.objects.all()
+        actor = get_object_or_404(actors, pk=pk)
```

Comment:
instead you should add this to the controller 
ALSO it is not a good idea to get all the objects then call get_object_or_404()
instead you should use 
Actor.objects.filtre(id=pk).first()

the idea of getting all objects will cost a lot  "loading all the data then filter" 
instead do the filter on the database